src/org/folio/rest_v2/Eureka.groovy

```diff
@@ -200,4 +200,51 @@ class Eureka extends Common {
       logger.error("Not able to assign ${resourceIds} to ${roleId}")
     }
   }
+
+  boolean isApplicationRegistered(String applicationId) {
+
+    String url = generateKongUrl("/applications/${applicationId}")
+
+    try {
+      restClient.get(url).body
+      logger.info("Application ${applicationId} is already registered.")
+      return true
+    } catch (RequestException e) {
+      if (e.statusCode == HttpURLConnection.HTTP_NOT_FOUND) {
+        logger.info("Application id ${applicationId} not found in Application manager. Proceeding with registration.")
+        return false
+      } else {
+        throw new RequestException("Application manager is unavailable", e.statusCode)
+      }
+    }
+  }
+
+  def getDescriptorsList(applicationId) {
+
+    steps.awscli.withAwsClient() {
+      steps.sh(script: "aws s3api get-object --bucket ${Constants.EUREKA_BUCKET_NAME} --key apps/${applicationId}.json ${applicationId}.json")
+    }
+    logger.info(steps.readJSON(file: "${applicationId}.json"))
+    return steps.readJSON(file: "${applicationId}.json")
+  }
+
+  def registerApplication(String applicationId) {
+    String descriptorsList = getDescriptorsList(applicationId)
+    if (isApplicationRegistered(applicationId)) {
+      logger.warning("Application ${applicationId} is already registered.")
+      return
+    }
+
+    String url = "${Constants.EUREKA_KONG_URL}applications?check=false"
```

Comment:
This URL can be taken from 'EurekaTenant.kongUrl' attribute

Reply:
Fixed